File: app/src/main/java/com/buzzybrains/mvvmarchitecture/data/NoteDao.java
package com.buzzybrains.mvvmarchitecture.data;


import com.buzzybrains.mvvmarchitecture.model.Note;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface NoteDao {

    @Insert
    long insert(Note note);


    @Update
    void update(Note note);


    @Delete
    void delete(Note note);


    @Query("DELETE from note_table")
    void deleteAllNotes();


    @Query("SELECT *  from note_table ORDER by priority desc")
    LiveData<List<Note>> getAllNotes();

    @Query("SELECT *  from note_table where isSync=0")
    List<Note> getAsyncNotes();


}

File: app/src/main/java/com/buzzybrains/mvvmarchitecture/data/NoteDao.kt
package com.buzzybrains.mvvmarchitecture.data


import com.buzzybrains.mvvmarchitecture.model.Note

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface NoteDao {


    @get:Query("SELECT *  from note_table ORDER by priority desc")
    val allNotes: LiveData<List<Note>>

    @get:Query("SELECT *  from note_table where isSync=0")
    val asyncNotes: List<Note>

    @Insert
    fun insert(note: Note): Long


    @Update
    fun update(note: Note)


    @Delete
    fun delete(note: Note)


    @Query("DELETE from note_table")
    fun deleteAllNotes()


}
